show-tech/roles/cores/files/handle_cores.py

```diff
@@ -0,0 +1,138 @@
+import glob
+import gzip
+import logging
+import json
+import os
+import shutil
+import subprocess
+import shlex
+
+logging.basicConfig(format='%(levelname)s: %(message)s' ,level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class ComponentCores(object):
+    def __init__(self, cores_map, component, max_age, dest_dir):
+        self.component = component
+        self.component_data = cores_map[component]
+        self.max_age = max_age
+        self.dest_dir = dest_dir
+        self.all_cores = glob.glob(self.component_data["path"])
+        self.cores = self.filter_files_by_ctime()
+        self.app_binary = self.component_data["binary"]
+        self.core_dirs = { os.path.dirname(x) for x in self.cores }
+
+    def filter_files_by_ctime(self):
+        cores = []
+        start_time = (datetime.datetime.now() - datetime.timedelta(days=self.max_age)).timestamp()
+        for corefile in self.all_cores:
+            if os.path.getctime(corefile) > start_time:
+                cores.append(corefile)
+        return cores
+
+    def get_core_files(self):
+        return self.cores
+
+    def get_core_dirs(self):
+        return self.core_dirs
+
+    def process_cores(self):
+        # Copy them to destination folder
+        logger.debug("Processing cores of component {} dirs {}".format(self.component, self.get_core_dirs()))
+        for core_dir in self.get_core_dirs():
+            dest_core_dir = os.path.join(self.dest_dir, os.path.basename(core_dir))
+            logger.info("Copying {} to {}".format(core_dir, dest_core_dir))
+            if os.path.exists(dest_core_dir):
+                shutil.rmtree(dest_core_dir)
+            shutil.copytree(core_dir, dest_core_dir)
+        # Uncompress them on source dir
+        for core_file in  self.get_core_files():
+            logger.debug("Analyzing {}".format(core_file))
+            core = CoreFile(core_file, self.app_binary, self.dest_dir)
+            core.analyze()
+
+
+class CoreFile(object):
+    def __init__(self, core_file_name, app_binary, dest_dir):
+        self.core_file_name = core_file_name
+        self.uncompressed_core_file = core_file_name
+        self.app_binary = app_binary
+        self.dest_dir = dest_dir
+
+    def uncompress_file(self):
+        if self.is_compressed():
+            uncompressed_core_file = self.core_file_name.replace(".gz", "")
+            with gzip.open(self.core_file_name, 'rb') as f_in:
+                with open(uncompressed_core_file, 'wb') as f_out:
+                    shutil.copyfileobj(f_in, f_out)
+            self.uncompressed_core_file = uncompressed_core_file
+            os.remove(self.core_file_name)
+
+    def is_compressed(self):
+        if self.core_file_name.endswith(".gz"):
+            return True
+        return False
+
+    def analyze(self):
+        self.uncompress_file()
+        cmd = "gdb --batch --quiet -ex 'start thread apply all bt full' -ex 'quit'  {} {}".format(self.app_binary, self.uncompressed_core_file)
```

Comment:
might be useful to dump other state as well including `info globals, info locals, info args`

Reply:
Does full dump this information ?